src/hmem_cuda.c

```diff
@@ -49,12 +49,57 @@ struct cuda_ops {
 	const char *(*cudaGetErrorString)(cudaError_t error);
 };
 
+#if defined(ENABLE_CUDA_DLOPEN) && defined(HAVE_LIBDL)
+
+#include <dlfcn.h>
+
+struct cuda_ops_name {
+	const char *cudaMemcpy;
+	const char *cudaGetErrorName;
+	const char *cudaGetErrorString;
+};
+
+static const char *cudart_filename = "libcudart.so";
+static void *cudart_handle;
+
+static cudaError_t stub_cudaMemcpy(void *dst, const void *src, size_t count,
+				   enum cudaMemcpyKind kind)
+{
+	return cudaErrorInvalidValue;
+}
+
+static const char *stub_cudaGetErrorName(cudaError_t error)
+{
+	return NULL;
+}
+
+static const char *stub_cudaGetErrorString(cudaError_t error)
+{
+	return NULL;
+}
+
+static struct cuda_ops cuda_ops = {
+	.cudaMemcpy = stub_cudaMemcpy,
+	.cudaGetErrorName = stub_cudaGetErrorName,
+	.cudaGetErrorString = stub_cudaGetErrorString,
+};
```

Comment:
Why do we need these stub functions, rather than just setting them to NULL?  If we can't set them during initialization, init fails, and everything aborts at that point.  If a provider tries to use them, that's a coding error, and we'll just crash on a null deref, which will give us a simple stack trace pointing directly to the issue.

Reply:
Will fix.